### ISSUE TITLE
CMake dependency on Kokkos Kernels needs to be updated

### ISSUE THREAD

**Issue by PhilMiller** [DARMA-tasking/checkpoint#263]
When building EMPIRE against newer Trilinos with Kokkos 3.7, I get the following warning:

```
CMake Warning (dev) at TPL/checkpoint/src/CMakeLists.txt:86 (target_link_libraries):
  The library that is being linked to, kokkoskernels, is marked as being
  deprecated by the owner.  The message provided by the developer is:

  WARNING: The non-namespaced target 'kokkoskernels' is deprecated! If always
  using newer versions of the project 'Trilinos', then use the new namespaced
  target 'KokkosKernels::kokkoskernels', or better yet,
  'KokkosKernels::all_libs' to be less sensitive to changes in the definition
  of targets in the package 'KokkosKernels'.  Or, to maintain compatibility
  with older or newer versions the project 'Trilinos', instead link against
  the libraries specified by the variable 'KokkosKernels_LIBRARIES'.
```

**Comment by PhilMiller** [comment]
Per discussion with the Kokkos folks, there's currently no way to work around this that will suit both stand-alone and Trilinos-embedded Kokkos Kernels. Hopefully a fix will make it into Kokkos Kernels 4.0.1

**Comment by PhilMiller** [comment]
Can fix after kokkos/kokkos-kernels#1749 makes its way into Trilinos and downstream

**Comment by PhilMiller** [comment]
With kokkos/kokkoskernels#2020 merged and now released in Kokkos/Kernels 4.2, the workaround may no longer be necessary.
